prow/cron/cron.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cron provides a wrapper of robfig/cron, which manages schedule cron jobs for horologium
+package cron
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/robfig/cron"
+
+	"k8s.io/test-infra/prow/config"
+)
+
+// jobStatus is a cache layer for tracking existing cron jobs
+type jobStatus struct {
+	entryID cron.EntryID
+	trigger bool
+	cronStr string
+}
+
+// Cron is a wrapper for cron.Cron
+type Cron struct {
+	cronAgent *cron.Cron
+	jobs      map[string]*jobStatus
+	lock      sync.Mutex
+}
+
+// NewClient makes a new Cron object
+func NewClient() *Cron {
+	return &Cron{
+		cronAgent: cron.New(),
+		jobs:      map[string]*jobStatus{},
+	}
+}
+
+// Start kicks off current cronAgent scheduler
+func (c *Cron) Start() {
+	c.cronAgent.Start()
+}
+
+// Stop pauses current cronAgent scheduler
+func (c *Cron) Stop() {
+	c.cronAgent.Stop()
+}
+
+// QueuedJobs returns a map of jobs that need to be triggered
+// and reset trigger in jobStatus
+func (c *Cron) QueuedJobs() map[string]bool {
```

Comment:
Why not a slice?

Reply:
Horologium need to used it as a map (or a set), I guess I can return slice here and construct a map in horologium

Reply:
That sounds good.

Reply:
done